src/tests/unit/workload/test_search.py

```diff
@@ -42,97 +46,545 @@ def cbcsdk_mock(monkeypatch, cb):
+    assert facet.values[0].id == "NOT_INSTALLED"
+    assert facet.values[0].name == "NOT_INSTALLED"
+    assert facet.values[0].total == 45
+
+
+def test_facet_aws(cbcsdk_mock):
+    """Tests faceting AWS compute resources."""
+    def post_validate(url, body, **kwargs):
+        assert body == AWS_FACET_REQUEST
+        return AWS_FACET_RESPONSE
+
+    cbcsdk_mock.mock_request("POST", "/lcm/view/v2/orgs/test/compute_resources/_facet", post_validate)
+    api = cbcsdk_mock.api
+    query = api.select(AWSComputeResource).set_subnet_id(["alphaworx"])
+    results = query.facet(["auto_scaling_group_name", "cloud_provider_tags", "platform", "platform_details",
+                           "virtual_private_cloud_id"])
+    assert len(results) == 5
+    facet = results[0]
+    assert facet.field == "cloud_provider_tags"
+    assert len(facet.values) == 2
+    assert facet.values[0].id == "Name##CB-Installed-Oregon"
+    assert facet.values[0].name == "Name##CB-Installed-Oregon"
+    assert facet.values[0].total == 6
+    assert facet.values[1].id == "Name##CB-Installed-Wyoming"
+    assert facet.values[1].name == "Name##CB-Installed-Wyoming"
+    assert facet.values[1].total == 9
+    facet = results[1]
+    assert facet.field == "auto_scaling_group_name"
+    assert len(facet.values) == 2
+    assert facet.values[0].id == "Virginia-ASG"
+    assert facet.values[0].name == "Virginia-ASG"
+    assert facet.values[0].total == 5
+    assert facet.values[1].id == "Georgia-ASG"
+    assert facet.values[1].name == "Georgia-ASG"
+    assert facet.values[1].total == 10
+    facet = results[2]
+    assert facet.field == "virtual_private_cloud_id"
+    assert len(facet.values) == 1
+    assert facet.values[0].id == "vpc-abcd123"
+    assert facet.values[0].name == "vpc-abcd123"
+    assert facet.values[0].total == 15
+    facet = results[3]
+    assert facet.field == "platform_details"
+    assert len(facet.values) == 1
+    assert facet.values[0].id == "Linux/UNIX"
+    assert facet.values[0].name == "Linux/UNIX"
+    assert facet.values[0].total == 15
+    facet = results[4]
+    assert facet.field == "platform"
+    assert len(facet.values) == 1
+    assert facet.values[0].id == "Unix/Linux"
+    assert facet.values[0].name == "Unix/Linux"
+    assert facet.values[0].total == 15
+
+
+def test_download_resource(cbcsdk_mock):
+    """Test creating a download request for compute resources."""
+    def post_validate(url, body, **kwargs):
+        assert body == WORKLOAD_DOWNLOAD_REQUEST
+        return DOWNLOAD_RESPONSE
+
+    cbcsdk_mock.mock_request("POST", "/lcm/view/v2/orgs/test/compute_resources/_search/download", post_validate)
+    cbcsdk_mock.mock_request("GET", "/jobs/v1/orgs/test/jobs/120066", DOWNLOAD_JOB_RESPONSE)
+    api = cbcsdk_mock.api
+    query = api.select(ComputeResource).set_installation_status(["NOT_INSTALLED", "PENDING", "ERROR"])
+    query.sort_by("created_at", "DESC")
+    job = query.download("CSV")
+    assert job.id == 120066
```

Comment:
Can we assert that we get a job back instead of only verifying the id?

Reply:
Making this change.